File: Languages/Perl/rand_avg.pl
#!/usr/bin/perl

# Hello, World!
print "Hello, World!\n";

use strict;
use warnings;

sub generate_random_numbers {
    my $count = 1000;
    my $output_dir = "../rand_avg output";
    my $output_file = "$output_dir/random_numbers.txt";
    
    # Create output directory
    unless (-d $output_dir) {
        mkdir $output_dir, 0755 or die "Cannot create directory: $!";
    }
    
    # Generate 1000 random numbers
    my @random_numbers;
    my $sum = 0;
    
    for my $i (1..$count) {
        my $num = int(rand(1000));
        push @random_numbers, $num;
        $sum += $num;
    }
    
    # Calculate mean
    my $mean = $sum / $count;
    
    # Save to file
    open my $fh, '>', $output_file or die "Cannot open file: $!";
    for my $num (@random_numbers) {
        print $fh "$num\n";
    }
    close $fh;
    
    print "Generated 1000 random numbers\n";
    printf "Mean: %.2f\n", $mean;
    print "Saved to: $output_file\n";
}

generate_random_numbers();

File: Languages/Rust/rand_avg.rs
use std::fs;
use std::io::{self, Write};
use rand::Rng;

// Hello, World!
fn hello() {
    println!("Hello, World!");
}

fn generate_random_numbers() -> Result<(), Box<dyn std::error::Error>> {
    const COUNT: usize = 1000;
    const OUTPUT_DIR: &str = "../rand_avg output";
    const OUTPUT_FILE: &str = "../rand_avg output/random_numbers.txt";
    
    // Create output directory
    fs::create_dir_all(OUTPUT_DIR)?;
    
    // Generate 1000 random numbers
    let mut random_numbers = Vec::new();
    let mut rng = rand::thread_rng();
    let mut sum = 0.0;
    
    for _ in 0..COUNT {
        let num: i32 = rng.gen_range(0..1000);
        random_numbers.push(num);
        sum += num as f64;
    }
    
    // Calculate mean
    let mean = sum / COUNT as f64;
    
    // Save to file
    let mut file = fs::File::create(OUTPUT_FILE)?;
    for num in &random_numbers {
        writeln!(file, "{}", num)?;
    }
    
    println!("Generated 1000 random numbers");
    println!("Mean: {:.2}", mean);
    println!("Saved to: {}", OUTPUT_FILE);
    
    Ok(())
}

fn main() {
    hello();
    
    if let Err(e) = generate_random_numbers() {
        eprintln!("Error: {}", e);
    }
}
